### ISSUE TITLE
Validate - docs do not reflect latest schema for azurerm_data_factory_linked_service_data_lake_storage_gen2

### ISSUE THREAD

**Issue by drjwelch** [hashicorp/terraform-provider-azurerm#12180]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureRM Provider) Version

tf 0.14.7
azurerm 2.63
### Affected Resource(s)

`azurerm_data_factory_linked_service_data_lake_storage_gen2`

### Terraform Configuration Files

```
resource "azurerm_data_factory_linked_service_data_lake_storage_gen2" "datalake_ls" {
  name                  = "datalake_ls"               
  resource_group_name   = azurerm_resource_group.myRG.name
  data_factory_name     = azurerm_data_factory.myFactory.name
  use_managed_identity  = true                       
  tenant                = var.myTenantId
  url                   = "https://myDataLake.dfs.core.windows.net/"
}
```

### Expected Behaviour

Validation passes

### Actual Behaviour
```
Error: "tenant": all of `service_principal_id,tenant` must be specified
```

### Steps to Reproduce

1. `terraform validate`

### Important Factoids

Docs need to be updated

[Docs](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/data_factory_linked_service_data_lake_storage_gen2) state as follows:
`tenant` is required
`use_managed_identity` is incompatible with `service_principal_id` and `service_principal_key`

But schema conflicts saying `tenant` is optional and required with `service_principal_key` (which conflicts with `use_managed_identity`)

[Schema](https://github.com/terraform-providers/terraform-provider-azurerm/blob/f3649f9805f552bced319fabdbda0217b2797f0b/azurerm/internal/services/datafactory/data_factory_linked_service_data_lake_storage_gen2_resource.go) states:
```
"tenant": {
        				Type:          pluginsdk.TypeString,
        				Optional:      true,
        				ValidateFunc:  validation.StringIsNotEmpty,
        				RequiredWith:  []string{"service_principal_id"},
        				ConflictsWith: []string{"storage_account_key", "use_managed_identity"},
        			},
```
and
```
"service_principal_id": {
        				Type:          pluginsdk.TypeString,
        				Optional:      true,
        				ValidateFunc:  validation.IsUUID,
        				RequiredWith:  []string{"service_principal_key"},
        				ConflictsWith: []string{"storage_account_key", "use_managed_identity"},
        				AtLeastOneOf:  []string{"service_principal_key", "service_principal_id", "storage_account_key", "use_managed_identity"},
        			},
```

### References

None

**Comment by drjwelch** [comment]
For anyone landing here with validate suddenly failing, simply remove tenant from the resource definition

**Comment by favoretti** [comment]
Hi there, thank you for reporting this. Will you be willing to create a PR to fix the docs?

**Comment by drjwelch** [comment]
> 
> 
> Hi there, thank you for reporting this. Will you be willing to create a PR to fix the docs?

I'd be very happy to help but I'm not sure where to go to edit the docs; can you give me a hint?!

**Comment by favoretti** [comment]
Awesome! Fork the repo, edit the file https://github.com/terraform-providers/terraform-provider-azurerm/blob/master/website/docs/r/data_factory_linked_service_data_lake_storage_gen2.html.markdown in your fork, then commit changes to your fork, push them and create a pull request.

Looking forward to your upcoming contribution!

**Comment by drjwelch** [comment]
@favoretti I think I have completed this - please could you review draft pull request here - thanks
https://github.com/terraform-providers/terraform-provider-azurerm/pull/12196

**Comment by drjwelch** [comment]
@favoretti I saw a subsequent issue and PR on this - should I close this one?

**Comment by rcskosir** [comment]
This should be closed by #12359.
